clippy_lints/src/single_range_in_vec_init.rs

```diff
@@ -91,14 +92,17 @@ impl LateLintPass<'_> for SingleRangeInVecInit {
         };
 
         if cx.tcx.qpath_is_lang_item(qpath, LangItem::Range)
+            && inner_expr.span.is_desugaring(DesugaringKind::RangeExpr)
```

Comment:
Since you now check for a desugared expression, is the `LangItem` test above it still required?